classes/Validate.php

```diff
@@ -165,11 +170,16 @@ public static function isImageSize($size)
      */
     public static function isCustomerName($name)
     {
-        $validityPattern = Tools::cleanNonUnicodeSupport(
-            '/^(?:[^0-9!<>,;?=+()\/\\@#"°*`{}_^$%:¤\[\]|\.。]|[\.。](?:\s|$))*$/u'
+        $validatorBuilder = Validation::createValidatorBuilder();
+        $validatorBuilder->setConstraintValidatorFactory(
+            new CustomerNameValidatorFactory(new CharacterCleaner())
         );
+        $validator = $validatorBuilder->getValidator();
+        $violations = $validator->validate($name, [
+            new CustomerName(),
+        ]);
 
-        return preg_match($validityPattern, $name);
+        return (count($violations) !== 0) ? 0 : 1;
```

Comment:
`is` methods must return boolean.
```suggestion
        return count($violations) !== 0;
```

Reply:
Yeap, but it is in the legacy and the previous behavior was 0 or 1 and false or true. So I don't change this behavior.

Reply:
It sucks! :sweat_smile:

Reply:
It's the legacy :)